p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/repository/DefaultRepositoryIdManager.java

```diff
@@ -19,27 +19,38 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Stream;
 
-import org.codehaus.plexus.component.annotations.Component;
-import org.codehaus.plexus.component.annotations.Requirement;
-import org.codehaus.plexus.logging.Logger;
 import org.eclipse.tycho.IRepositoryIdManager;
 import org.eclipse.tycho.MavenRepositoryLocation;
 import org.eclipse.tycho.MavenRepositorySettings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
 
 /**
  * Helper class for the Remote*RepositoryManagers taking care of mapping repository URLs to the
  * settings.xml-configured mirrors and setting passwords.
  */
-@Component(role = IRepositoryIdManager.class)
+@Singleton
+@Named
 public class DefaultRepositoryIdManager implements IRepositoryIdManager {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    // TODO: wut? repositoryIds may CHANGE per project, as project may define new ones
```

Comment:
Maven wants server-id to get authenticaction / proxy infos, as P2 repository has no ID we let this component "learn" what URI was used in a given id context (e.g. pom or target file), so later we can map them back to ask if maven has credentials for that ID.

As credential IDs are global one can't use different credentials/mirrors/proxy per ID and even if in theory it could be different per project this actually won't work well